10up/plugin-check
# Implement `Abstract_File_Check` base class as well as two plugin requirement checks using it

### Description of the Change

* Implements an `Abstract_File_Check` base class to make it easy to implement checks like those used by the WordPress theme check plugin or by the early plugin review tool, as described in #174
* Implement two essential checks for minimum plugin requirements that use the new base class:
    * `Code_Obfuscation_Check`, based on https://github.com/WordPress/plugin-check/blob/trunk/checks/code-obfuscation.php
    * `File_Type_Check`, based on https://github.com/WordPress/plugin-check/blob/trunk/checks/file-checks.php
* Test coverage for the new checks is added, also indirectly covering the base class.

Closes #174

### How to test the Change
* Install a plugin that contains any of the following:
    * compressed files, e.g. ZIP, TAR, GZ, etc.
    * `.phar` files
    * version control checkouts, e.g. a `.git` or `.svn` folder
    * hidden files starting with `.`
    * obfuscated PHP files obfuscated with either Zend Guard, Source Guardian, or ionCube
* Ensure those plugin rule violations are flagged when running the plugin checker for that plugin.
* Alternatively to real plugins, you can also install/activate the relevant plugins in the `tests/phpunit/testdata/plugins` folder.

### Checklist:
